### PR TITLE
RBAC: Fix plugins pages access-control

### PR DESCRIPTION
**Pull request by gamab**
**What is this feature?**

A viewer user cannot access plugin pages. They systematically get redirected. Even if they have been granted: 
`fixed:plugins.app:reader`, `fixed:plugins:writer`,  `fixed:plugins:maintainer`.

It's because the frontend access-control check and the backend access-control middleware solely enforce checks on the user basic role (`Admin`, `ServerAdmin`), without checking permissions.

This PR intends to fix this and align access-control with what we do for the [navlink](https://github.com/grafana/grafana/blob/6184bf21e28e67b25f50b31e19f1e327efa78a92/pkg/services/navtree/navtreeimpl/admin.go#L21).

Fixes #

**Special notes for your reviewer:**

~~I worked a while back on this and left a [fixme](https://github.com/grafana/grafana/blob/6184bf21e28e67b25f50b31e19f1e327efa78a92/pkg/services/navtree/navtreeimpl/admin.go#L20). I don't remember what exactly was the issue with just enforcing access control :thinking:~~

~~Changing the `CanAdminPlugins` middleware also impacts some connections routes. We need to assess if that's a problem or not.~~
All good, double checked by the plugin squad, these are actually plugin related pages :+1: 

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/api/api.go [hunk 1]
```diff
@@ -124,18 +124,18 @@ func (hs *HTTPServer) registerRoutes() {
...
+	r.Get("/plugins/:id/", middleware.CanAdminPlugins(hs.Cfg, hs.AccessControl), hs.Index)
+	r.Get("/plugins/:id/edit", middleware.CanAdminPlugins(hs.Cfg, hs.AccessControl), hs.Index) // deprecated
+	r.Get("/plugins/:id/page/:page", middleware.CanAdminPlugins(hs.Cfg, hs.AccessControl), hs.Index)
 
 	r.Get("/connections/datasources", authorize(datasources.ConfigurationPageAccess), hs.Index)
 	r.Get("/connections/datasources/new", authorize(datasources.NewPageAccess), hs.Index)
 	r.Get("/connections/datasources/edit/*", authorize(datasources.EditPageAccess), hs.Index)
 	r.Get("/connections", authorize(datasources.ConfigurationPageAccess), hs.Index)
 	r.Get("/connections/add-new-connection", authorize(datasources.ConfigurationPageAccess), hs.Index)
-	r.Get("/connections/datasources/:id", middleware.CanAdminPlugins(hs.Cfg), hs.Index)
```
Comment (gamab) [inline pkg/api/api.go:137]: Having only the plugin middleware here seems a bit suspicious to me :thinking:
 
Isn't it going to prevent users from configuring data sources that are not related to a plugin?
Isn't it also going to allow users that don't have `datasources.ConfigurationPageAccess` to access the page?

Comment (gamab) [reply pkg/api/api.go:137]: In an ideal world we should not have to check plugin permissions when dealing with datasources.

**Review by IevaVasiljeva — APPROVED** [review]
🚢

File: pkg/api/api.go (hunk 1 above)
Comment (gamab) [reply pkg/api/api.go:137]: Issue that introduced this change: https://github.com/grafana/grafana/pull/61397

Comment (marefr) [reply pkg/api/api.go:137]: So this is basically the same page as plugin catalog showing the details of a plugin so think it make sense to have it as is. See below

Connections/Add new connection: 
image

Administration/Plugins:
image

> Isn't it going to prevent users from configuring data sources that are not related to a plugin?

Plugins and datasources are different and thereby different permissions or am I misunderstanding something?

Comment (leventebalogh) [reply pkg/api/api.go:137]: @marefr is correct, this route is actually showing the "plugin details" page for a datasource plugin, and the settings of a datasource are available on this route:

```go
r.Get("/connections/datasources/edit/*", authorize(datasources.EditPageAccess), hs.Index)
```

I think it should be fine as it is, however I do agree that it's misleading - maybe we can add a comment above it?

**Review by leventebalogh — APPROVED** [review]
Nice catch @gamab! 

I think the fix looks legit, and I couldn't think of any scenarios where it could go wrong. For connections: the changes in the `CanAdminPlugins` middleware shouldn't be a problem, as the pages that depend on the middleware are meant for managing the datasource plugins, and not for the datasource instances themselves.

File: pkg/api/api.go (hunk 1 above)
Comment (gamab) [reply pkg/api/api.go:137]: Ahhhh nice, thanks for the explanation! Really thought this was the edit page!
I'll add a comment :)
